napari/_qt/dialogs/qt_notification.py

```diff
@@ -68,6 +80,7 @@ class NapariQtNotification(QDialog):
     message: MultilineElidedLabel
     source_label: QLabel
     severity_icon: QLabel
+    dispatcher = _NotificationDispatcher()
```

Comment:
>I have big worries about this initialization on import. QObjects should not be initialized before QApplication creation. >Could you delay a constructor call? 

This works fine

```python
from qtpy.QtCore import Signal, QObject
from qtpy.QtWidgets import QApplication


class _NotificationDispatcher(QObject):
    sig_notified = Signal(object)


def test(obj):
    print(obj)


dispatcher = _NotificationDispatcher()
dispatcher.sig_notified.connect(test)
dispatcher.sig_notified.emit("Works!")

app = QApplication([])
app.exec_()
```

Reply:
Please see these docs https://doc.qt.io/qt-5/threads-qobject.html, especially fragment: 

> In general, creating QObjects before the QApplication is not supported and can lead to weird crashes on exit, depending on the platform. This means static instances of QObject are also not supported. A properly structured single or multi-threaded application should make the QApplication be the first created, and last destroyed QObject.

I suggest 

```python
class _QtMainWindow(QMainWindow):
    # This was added so that someone can patch
    # `napari._qt.qt_main_window._QtMainWindow._window_icon`
    # to their desired window icon
    _window_icon = NAPARI_ICON_PATH
	dispatcher = None 
	

    # To track window instances and facilitate getting the "active" viewer...
    # We use this instead of QApplication.activeWindow for compatibility with
    # IPython usage. When you activate IPython, it will appear that there are
    # *no* active windows, so we want to track the most recently active windows
    _instances: ClassVar[List['_QtMainWindow']] = []
    def __init__(self, qt_viewer: QtViewer, parent=None) -> None:
        super().__init__(parent)
		if self.dispatcher is None:
			self.__class__.dispatcher = _NotificationDispatcher()
```

Reply:
>Please see these docs https://doc.qt.io/qt-5/threads-qobject.html, especially fragment:

You are right @Czaki. Thanks for the input :)